### ISSUE TITLE
图神经网络

### ISSUE THREAD

**Issue by wangziyan2004** [wangziyan2004/GNN_Learning#2]

**Comment by wangziyan2004** [comment]
# GNN Learning Roadmap

## 1. GNN任务
- [ ] 节点分类
- [ ] 链接预测
- [ ] 图分类
- [ ] 图回归
- [ ] Node-level / Edge-level / Graph-level

## 2. PyTorch Geometric
- [ ] Data
- [ ] Dataset
- [ ] DataLoader
- [ ] x
- [ ] edge_index
- [ ] y
- [ ] train_mask
- [ ] val_mask
- [ ] test_mask

## 3. GNN实现
- [ ] GCNConv
- [ ] SAGEConv
- [ ] GATConv
- [ ] Cora + GCN
- [ ] Cora + GraphSAGE
- [ ] Cora + GAT

## 4. GNN实验
- [ ] Loss
- [ ] Optimizer
- [ ] Accuracy
- [ ] Precision
- [ ] Recall
- [ ] F1
- [ ] Epoch
- [ ] Learning Rate
- [ ] Early Stopping
- [ ] Hyperparameter

## 5. GNN核心问题
- [ ] Over-smoothing
- [ ] Over-squashing
- [ ] GNN深度问题
- [ ] GNN表达能力

## 6. 经典论文
- [ ] GCN
- [ ] GraphSAGE
- [ ] GAT

## 7. 图级任务
- [ ] Global Mean Pooling
- [ ] Global Max Pooling
- [ ] Global Sum Pooling
- [ ] Hierarchical Pooling
- [ ] Graph Classification

## 8. Graph Transformer
- [ ] Self-Attention
- [ ] GAT vs Transformer
- [ ] Graph Transformer

## 9. GNN进阶
- [ ] 异质图
- [ ] 动态图
- [ ] 图对比学习
- [ ] Graph Transformer

## 10. 科研
- [ ] 阅读论文
- [ ] 复现论文
- [ ] 实验设计
- [ ] Baseline
- [ ] Ablation Study
- [ ] 找研究问题
- [ ] 尝试改进
